File: JavaScript_Courses/JavaScript_Basics/Preparation/03_Conditional_Statements_Advanced_Exercise/09_Ski_Trip.js
function skiTrip(input) {
    let nights = Number(input[0]) - 1;
    let space = input[1];
    let valuation = input[2];

    let totalCost = 0;

    if (space == "room for one person") {
        totalCost = nights * 18;
    } else if (space == "apartment") {
        totalCost = nights * 25;
        if (nights < 10) {
            totalCost *= 0.70;
        } else if (nights <= 15) {
            totalCost *= 0.65;
        } else {
            totalCost *= 0.50;
        }
    } else if (space == "president apartment") {
        totalCost = nights * 35;
        if (nights < 10) {
            totalCost *= 0.90;
        } else if (nights <= 15) {
            totalCost *= 0.85;
        } else {
            totalCost *= 0.80;
        }
    }

    if (valuation == "positive") {
        totalCost *= 1.25;
    } else if (valuation == "negative") {
        totalCost *= 0.90;
    }

    console.log(`${totalCost.toFixed(2)}`);
}

// skiTrip(["14", "apartment", "positive"]);

File: Python_Courses/Python_Basics/Python_PB_2023_10/03_Conditional_Statements_Advanced_Exercise/09_Ski_Trip.py
days = int(input())
type_stay = input()  # "room for one person", "apartment" или "president apartment"
evaluation = input()  # "positive"  или "negative"

nights = days - 1
cost_per_day = 0
discount = 0

if type_stay == "room for one person":
    cost_per_day = 18.00
    if days < 10:
        discount = 0
    elif days <= 15:  # [10; 15]
        discount = 0
    else:
        discount = 0
elif type_stay == "apartment":
    cost_per_day = 25.00
    if days < 10:
        discount = 0.30
    elif days <= 15:  # [10; 15]
        discount = 0.35
    else:
        discount = 0.50
elif type_stay == "president apartment":  # else:
    cost_per_day = 35.00
    if days < 10:
        discount = 0.10
    elif days <= 15:  # [10; 15]
        discount = 0.15
    else:
        discount = 0.20

total_cost = nights * cost_per_day * (1 - discount)  # (days - 1) * ...
if evaluation == "positive":
    total_cost *= 1.25
    # total_cost = total_cost * (1 + 0.25)
elif evaluation == "negative":
    total_cost *= 0.90

print(f"{total_cost :.2f}")
